src/transports/dash/segment_parser.ts

```diff
@@ -69,18 +70,28 @@ export default function parser({ content,
                                                             0),
                           appendWindow: [period.start, period.end] });
   } else { // it is an initialization segment
+    const { privateInfos } = segment;
+    const shouldExtractCompleteInitChunk = privateInfos !== undefined &&
+                                           privateInfos.shouldGuessInitRange === true;
+    const completeInitChunk = shouldExtractCompleteInitChunk ?
+      extractCompleteInitChunk(chunkData) : chunkData;
+
+    if (completeInitChunk === -1) {
```

Comment:
Do not tell me that it's for that reason that you made `extractCompleteInitChunk` return `-1` :p